File: src/main/java/com/gxsneaker/gxsneaker/service/GioHangService.java
package com.gxsneaker.gxsneaker.service;

import com.gxsneaker.gxsneaker.entity.GioHang;
import com.gxsneaker.gxsneaker.entity.GioHangChiTiet;

public interface GioHangService {
    GioHang getGioHangByKhachHangId(Integer khachHangId);
    GioHangChiTiet addItemToCart(Integer khachHangId, Long chiTietSanPhamId, Integer soLuong);
    GioHangChiTiet updateItemQuantity(Integer cartItemId, Integer soLuong);
    void removeItemFromCart(Integer cartItemId);
    void clearCart(Integer khachHangId);
}

File: gx-sneaker-ui/src/services/gioHangService.js
import axios from "axios"

const API = "http://localhost:8080/api/gio-hang"

export const getCartByKhachHangId = (khachHangId) => axios.get(`${API}/khach-hang/${khachHangId}`)
export const addItemToCart = (khachHangId, chiTietSanPhamId, soLuong) => 
    axios.post(`${API}/khach-hang/${khachHangId}/add?chiTietSanPhamId=${chiTietSanPhamId}&soLuong=${soLuong}`)
export const updateItemQuantity = (cartItemId, soLuong) => 
    axios.put(`${API}/item/${cartItemId}?soLuong=${soLuong}`)
export const removeItemFromCart = (cartItemId) => axios.delete(`${API}/item/${cartItemId}`)
export const clearCart = (khachHangId) => axios.delete(`${API}/khach-hang/${khachHangId}/clear`)
